Do not delete dashboard uid fields

gw0 opened the pull request:
The purpose of the `uid` field is to allow migrating dashboards between Grafana instances and provisioning Grafana from configuration without breaking references. From my understanding this field is not instance-specific (such as `id`), therefore it should be preserved.

- https://grafana.com/docs/grafana/latest/administration/provisioning/#reusable-dashboard-urls

gw0 committed:
Do not delete dashboard uid field

gw0 commented:
I believe this also causes weird folder issues (a duplicate gets placed into `General`) and consequently there are issues with multiple slugs (because the slug-based Grafana API is being used).

trotttrotttrott reviewed (commented):
Change makes sense to me, but it's causing some acceptance tests to fail (really need to add those to CI).

gw0 committed:
Update tests to retain the dashboard uid field

gw0 commented:
You are right. I rebased and updated the acceptance tests to include the `uid` field. Its value should be retained if everything works correctly.

Now only the acceptance test that was failing before this PR is still failing:

```
$ GRAFANA_URL=http://172.17.0.1:3000 GRAFANA_AUTH=admin:admin make testacc
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v  -timeout 120m
?   	github.com/terraform-providers/terraform-provider-grafana	[no test files]
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_impl
--- PASS: TestProvider_impl (0.00s)
=== RUN   TestAccAlertNotification_basic
--- PASS: TestAccAlertNotification_basic (0.20s)
=== RUN   TestAccAlertNotification_invalid_frequence
    TestAccAlertNotification_invalid_frequence: testing.go:561: Step 0, expected error:
        
        errors during apply: time: invalid duration hi
        
        To match:
        
        invalid duration "hi"
        
        
--- FAIL: TestAccAlertNotification_invalid_frequence (0.04s)
=== RUN   TestAccAlertNotification_reminder_no_frequence
--- PASS: TestAccAlertNotification_reminder_no_frequence (0.04s)
=== RUN   TestAccDashboard_basic
--- PASS: TestAccDashboard_basic (0.31s)
=== RUN   TestAccDashboard_folder
--- PASS: TestAccDashboard_folder (0.34s)
=== RUN   TestAccDashboard_disappear
--- PASS: TestAccDashboard_disappear (0.15s)
=== RUN   TestAccDataSource_basic
--- PASS: TestAccDataSource_basic (2.14s)
=== RUN   TestAccFolderPermission_basic
--- PASS: TestAccFolderPermission_basic (0.54s)
=== RUN   TestAccFolder_basic
--- PASS: TestAccFolder_basic (0.23s)
=== RUN   TestAccOrganization_basic
--- PASS: TestAccOrganization_basic (0.46s)
=== RUN   TestAccOrganization_users
--- PASS: TestAccOrganization_users (0.71s)
=== RUN   TestAccOrganization_defaultAdmin
--- PASS: TestAccOrganization_defaultAdmin (0.34s)
=== RUN   TestAccTeamPreferences_basic
--- PASS: TestAccTeamPreferences_basic (0.50s)
=== RUN   TestAccTeam_basic
--- PASS: TestAccTeam_basic (0.34s)
=== RUN   TestAccTeam_Members
--- PASS: TestAccTeam_Members (0.40s)
=== RUN   TestAccUser_basic
--- PASS: TestAccUser_basic (0.39s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-grafana/grafana	7.151s
FAIL
GNUmakefile:16: recipe for target 'testacc' failed
make: *** [testacc] Error 1
```

trotttrotttrott reviewed (approved):
Thanks!

ghmeier commented:
Hi! Does this mean that the `uid` field is now required? This causes all resources to be recreated on every apply unless the `uid` is specified @trotttrotttrott.
